### PR TITLE
release docs

### LINKED ISSUE #494: Run through a new release of GLBC and Document the process

### PR DESCRIPTION
**Pull request by philbrookes**
Closes #494

## Description of Changes
Add docs detailing how to release GLBC stable to KCP stable.

### CODE REVIEW

File: docs/release.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+```
+git tag --sign --message vx.y.z vx.y.z upstream/main
+git push upstream vx.y.z
+```
+
+Create the release branch:
+```
+got checkout -b release-x.y.z
+git push upstream release-x.y.z
+```
```
Comment (mikenairn) [inline docs/release.md:29]: Might be worth pointing out you only need to do this for new minor releases see https://github.com/kcp-dev/kcp/blob/e798d372575e694db66bff61bd46700b413e1b9e/docs/content/en/main/concepts/developers/releasing.md#if-its-a-new-minor-version

File: docs/release.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+Create the signed tag (replace x.y.z with the release version):
+```
+git tag --sign --message vx.y.z vx.y.z upstream/main
+git push upstream vx.y.z
+```
+
+Create the release branch:
+```
+got checkout -b release-x.y.z
+git push upstream release-x.y.z
```
Comment (mikenairn) [inline docs/release.md:28]: This should only be release-x.y, patch release will be created from the release branch

**Comment by david-martin** [comment]
@philbrookes if consuming a release upstream, would you have to build your own image from the release tag?

**Comment by mikenairn** [comment]
> @philbrookes if consuming a release upstream, would you have to build your own image from the release tag?

Yes, until we merge https://github.com/kcp-dev/kcp-glbc/pull/491, but it probably needs tested.

**Comment by philbrookes** [comment]
> > @philbrookes if consuming a release upstream, would you have to build your own image from the release tag?
> 
> Yes, until we merge #491, but it probably needs tested.

Yes, I was assuming that would be dealt with by 0.3, but I can add a note here about doing that, if we feel it is necessary?

**Review by maleck13 — APPROVED** [review]

**Comment by maleck13** [comment]
@mikenairn happy to merge now?

**Review by mikenairn — APPROVED** [review]

**Pull request merged**